Tests/Common/Service/HttpEndpointTest.swift

```diff
@@ -44,4 +44,29 @@ class HttpEndpointTest: UnitTest {
 
         XCTAssertEqual(actual, expected)
     }
+
+    func test_getUrlString_givenRandomPathNeedsEncoding_expectEncodedPath() {
+        let identifierWithSpecialChar = String.randomStringWithSpecialCharacters()
+        let endpoint = CIOApiEndpoint.identifyCustomer(identifier: identifierWithSpecialChar)
+
+        let expectedIdentifier = identifierWithSpecialChar.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed) ?? identifierWithSpecialChar
+        let expected = "https://customer.io/api/v1/customers/\(expectedIdentifier)"
+
+        setHttpBaseUrls(trackingApi: "https://customer.io")
+
+        let actual = endpoint.getUrlString(baseUrls: httpBaseUrls)
+
+        XCTAssertEqual(actual, expected)
+    }
+
+    func test_getUrl_givenUnencodedPathWithSpecialCharacter_expectValidUrl() {
+        let identifierWithSpecialChar = "social-login|1234567890abcde"
+        let endpoint = CIOApiEndpoint.identifyCustomer(identifier: identifierWithSpecialChar)
+
+        setHttpBaseUrls(trackingApi: "https://customer.io")
+
+        let actualUrl = endpoint.getUrl(baseUrls: httpBaseUrls)
+
+        XCTAssertNotNil(actualUrl, "Expected valid URL but got nil. Ensure path is encoded correctly.")
```

Comment:
I prefer that we do a AssertEquals() here with a hard-coded URL that is generated. Would help me feel more confident in this function. Also, gives good examples for devs reading this code's function to understand it's behavior.

Reply:
I believe that using `assertequals` will increase the quality of the test for us to have more confidence in it. AssertNil makes sense in that we want to try and prevent a behavior the SDK has today. My suggestion is because I believe we can take this test function a step further and make the test even better then it is.

Reply:
I understand that but we are doing exactly that in the previous test, this test is to ensure we don't get a `nil`, which was the case for this behavior that has special characters produced `nil` as a URL. 

Now we have two tests, first, that asserts equal of making sure we get the URL we expect and the second to ensure having characters doesn't cause nil.